econml/tests/test_driv.py

```diff
@@ -1,161 +1,177 @@
+
+            est_list = [
+                DRIV(
+                    flexible_model_effect=StatsModelsLinearRegression(fit_intercept=False),
+                    model_final=StatsModelsLinearRegression(
+                        fit_intercept=False
+                    ),
+                    fit_cate_intercept=True,
+                    projection=projection,
+                    discrete_instrument=binary_Z,
+                    discrete_treatment=binary_T,
+                    featurizer=featurizer,
+                ),
+                LinearDRIV(
+                    flexible_model_effect=StatsModelsLinearRegression(fit_intercept=False),
+                    fit_cate_intercept=True,
+                    projection=projection,
+                    discrete_instrument=binary_Z,
+                    discrete_treatment=binary_T,
+                    featurizer=featurizer,
+                ),
+                SparseLinearDRIV(
+                    flexible_model_effect=StatsModelsLinearRegression(fit_intercept=False),
+                    fit_cate_intercept=True,
+                    projection=projection,
+                    discrete_instrument=binary_Z,
+                    discrete_treatment=binary_T,
+                    featurizer=featurizer,
+                ),
+                ForestDRIV(
+                    flexible_model_effect=StatsModelsLinearRegression(fit_intercept=False),
+                    projection=projection,
+                    discrete_instrument=binary_Z,
+                    discrete_treatment=binary_T,
+                    featurizer=featurizer,
+                ),
+            ]
+
+            if X is None:
+                est_list = est_list[:-1]
+
+            if binary_T and binary_Z:
+                est_list += [
+                    IntentToTreatDRIV(
+                        flexible_model_effect=StatsModelsLinearRegression(
+                            fit_intercept=False
+                        ),
+                        fit_cate_intercept=True,
+                        featurizer=featurizer,
+                    ),
+                    LinearIntentToTreatDRIV(
+                        flexible_model_effect=StatsModelsLinearRegression(
+                            fit_intercept=False
+                        ),
+                        featurizer=featurizer,
+                    ),
+                ]
+
+            for est in est_list:
+                with self.subTest(d_w=d_w, d_x=d_x, binary_T=binary_T,
+                                  binary_Z=binary_Z, projection=projection, featurizer=featurizer,
+                                  est=est):
+
+                    # TODO: serializing/deserializing for every combination -- is this necessary?
+                    # ensure we can serialize unfit estimator
+                    # pickle.dumps(est)
+
+                    est.fit(y, T, Z=Z, X=X, W=W)
+
+                    # ensure we can serialize fit estimator
+                    # pickle.dumps(est)
```

Comment:
We should always ensure we can serialize the fit estimator - different settings will result in different attributes being set (e.g. for DML, with a discrete treatment we create a one-hot encoder, and we also combine this with the featurizer if that is not None)

```suggestion
                    pickle.dumps(est)
```

Reply:
Note that we are not keeping the result around, we just verify that calling `pickle.dumps` does not fail, so hopefully this should not result in too much extra memory usage since the resulting string should be eligible for garbage collection right away.